lib/publiccloud/provider.pm

```diff
@@ -574,18 +574,15 @@ sub terraform_apply {
     # when testing instances that have nvidia gpus,
     # the zone (i.e. "sub-region") might not have them available and
     # suggest other zones instead (the pattern is GCE specific)
-    if ($ret != 0 && get_var('PUBLIC_CLOUD_NVIDIA') && ($tf_apply_output =~ /A .* VM instance with 1 .* accelerator\(s\) is currently unavailable in the .* zone\. Consider trying your request in the (.*) zone\(s\)/)) {
-        # split captured suggestions by a ',' char, trim whitespace
-        my @alternative_zones = split /\s*,\s*/, $1;
+    if ($ret != 0 && is_gce() && ($tf_apply_output =~ /A .* VM instance with 1 .* accelerator\(s\) is currently unavailable in the .* zone|Machine type with name .* does not exist in zone .*/)) {
```

Comment:
You're dropping the check for `PUBLIC_CLOUD_NVIDIA`.  Is this intended?

Reply:
yes , from now on same logic will be used not only for NVIDIA tests but for ANY GCE test because additionally to nvidia case we have more cases now when same logic is really needed ...